warehouse/query-core/src/main/java/datawave/query/jexl/visitors/TreeFlatteningRebuildingVisitor.java

```diff
@@ -228,34 +235,39 @@ protected ASTReferenceExpression advanceReferenceExpression(JexlNode jexlNode) {
         return null;
     }
     
-    protected JexlNode flattenTree(JexlNode currentNode, JexlNode newNode, Object data) {
+    protected void flattenTree(JexlNode currentNode, JexlNode newNode) {
         if (!currentNode.getClass().equals(newNode.getClass())) {
             log.warn("newNode is not the same type as currentNode ... something has probably gone horribly wrong");
         }
         
-        for (int i = 0; i < currentNode.jjtGetNumChildren(); i++) {
-            JexlNode node = (JexlNode) currentNode.jjtGetChild(i).jjtAccept(this, null);
-            JexlNode dereferenced = JexlASTHelper.dereference(node);
-            
-            if (acceptableNodesToCombine(currentNode, dereferenced, !node.equals(dereferenced))) {
-                flattenTree(dereferenced, newNode, data);
```

Comment:
By eliminating the recursion here, are we now  failing to flatten the subtrees?

Reply:
Yes, I added an additional test with additional nesting.

Reply:
I bet it will fail to flatten the AND tree within the OR tree however.  This test works with the current code:

    @Test
    public void multipleNestingTest() throws Exception {
        String query = "((a && (b && (c && d))) || b || (c || d || e || (f || g || (h || i || (((j || k)))))))";
        String expected = "((a && b && c && d) || b || c || d || e || f || g || h || i || j || k)";
        JexlNode node = TreeFlatteningRebuildingVisitor.flatten(JexlASTHelper.parseJexlQuery(query));
        Assert.assertEquals(expected, JexlStringBuildingVisitor.buildQuery(node));
    }

Reply:
It flattens, it is only the OR order that is different. I will address the ordering.

Reply:
Seems like maybe Ivan's example above would be a good test to add.